src/Middleware/InitStateMiddleware.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace TractorCow\Fluent\Middleware;
+
+use SilverStripe\Admin\AdminRootController;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\Middleware\HTTPMiddleware;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Injector\Injector;
+use TractorCow\Fluent\Extension\FluentDirectorExtension;
+use TractorCow\Fluent\Model\Domain;
+use TractorCow\Fluent\State\FluentState;
+
+/**
+ * InitStateMiddleware initialises the FluentState object and sets the current request locale and domain to it
+ */
+class InitStateMiddleware implements HTTPMiddleware
+{
+    use Configurable;
+
+    /**
+     * @var HTTPRequest
+     */
+    protected $request;
```

Comment:
Instead of making this a state, let's make the middleware stateless, and pass in $request argument to each method. That way the middleware is more re-usable across requests (i.e. testing).